File: SDKs/zoom.us/2.0.0/typescript/src/sdk/models/operations/getaccountbillinginvoice.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetAccountBillingInvoicePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=accountId" })
  accountId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=invoiceId" })
  invoiceId: string;
}


export class GetAccountBillingInvoice200ApplicationJsonInvoiceItems extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=charge_name" })
  chargeName?: string;

  @SpeakeasyMetadata({ data: "json, name=charge_number" })
  chargeNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=charge_type" })
  chargeType?: string;

  @SpeakeasyMetadata({ data: "json, name=end_date" })
  endDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=quantity" })
  quantity?: number;

  @SpeakeasyMetadata({ data: "json, name=start_date" })
  startDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=tax_amount" })
  taxAmount?: number;

  @SpeakeasyMetadata({ data: "json, name=total_amount" })
  totalAmount?: number;
}


export class GetAccountBillingInvoice200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=balance" })
  balance?: number;

  @SpeakeasyMetadata({ data: "json, name=currency" })
  currency?: string;

  @SpeakeasyMetadata({ data: "json, name=due_date" })
  dueDate?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=invoice_date" })
  invoiceDate?: string;

  @SpeakeasyMetadata({ data: "json, name=invoice_items", elemType: GetAccountBillingInvoice200ApplicationJsonInvoiceItems })
  invoiceItems?: GetAccountBillingInvoice200ApplicationJsonInvoiceItems[];

  @SpeakeasyMetadata({ data: "json, name=invoice_number" })
  invoiceNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: string;

  @SpeakeasyMetadata({ data: "json, name=target_date" })
  targetDate?: string;

  @SpeakeasyMetadata({ data: "json, name=tax_amount" })
  taxAmount?: number;

  @SpeakeasyMetadata({ data: "json, name=total_amount" })
  totalAmount?: number;
}


export class GetAccountBillingInvoiceRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetAccountBillingInvoicePathParams;
}


export class GetAccountBillingInvoiceResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getAccountBillingInvoice200ApplicationJSONObject?: GetAccountBillingInvoice200ApplicationJson;
}

File: SDKs/zoom.us/2.0.0/python/src/sdk/models/operations/getaccountbillinginvoice.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclasses.dataclass
class GetAccountBillingInvoicePathParams:
    account_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'accountId', 'style': 'simple', 'explode': False }})
    invoice_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'invoiceId', 'style': 'simple', 'explode': False }})
    

@dataclass_json
@dataclasses.dataclass
class GetAccountBillingInvoice200ApplicationJSONInvoiceItems:
    charge_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('charge_name') }})
    charge_number: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('charge_number') }})
    charge_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('charge_type') }})
    end_date: Optional[date] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('end_date'), 'encoder': utils.dateisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    quantity: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('quantity') }})
    start_date: Optional[date] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('start_date'), 'encoder': utils.dateisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    tax_amount: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tax_amount') }})
    total_amount: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('total_amount') }})
    

@dataclass_json
@dataclasses.dataclass
class GetAccountBillingInvoice200ApplicationJSON:
    balance: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('balance') }})
    currency: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('currency') }})
    due_date: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('due_date') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    invoice_date: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('invoice_date') }})
    invoice_items: Optional[list[GetAccountBillingInvoice200ApplicationJSONInvoiceItems]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('invoice_items') }})
    invoice_number: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('invoice_number') }})
    status: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    target_date: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('target_date') }})
    tax_amount: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tax_amount') }})
    total_amount: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('total_amount') }})
    

@dataclasses.dataclass
class GetAccountBillingInvoiceRequest:
    path_params: GetAccountBillingInvoicePathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetAccountBillingInvoiceResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    body: Optional[bytes] = dataclasses.field(default=None)
    get_account_billing_invoice_200_application_json_object: Optional[GetAccountBillingInvoice200ApplicationJSON] = dataclasses.field(default=None)
    
